monitor/flowgen/flowgen.go

```diff
@@ -2,25 +2,174 @@ package flowgen
+	"math/big"
 	"time"
 
-	"github.com/omni-network/omni/lib/tokens"
+	"github.com/omni-network/omni/e2e/app/eoa"
+	"github.com/omni-network/omni/lib/contracts"
+	"github.com/omni-network/omni/lib/contracts/solvernet"
+	"github.com/omni-network/omni/lib/errors"
+	"github.com/omni-network/omni/lib/ethclient/ethbackend"
+	"github.com/omni-network/omni/lib/evmchain"
+	"github.com/omni-network/omni/lib/log"
+	"github.com/omni-network/omni/lib/netconf"
+	"github.com/omni-network/omni/lib/xchain"
+	"github.com/omni-network/omni/monitor/flowgen/bridging"
+	"github.com/omni-network/omni/monitor/flowgen/types"
+	"github.com/omni-network/omni/monitor/flowgen/util"
+
+	"github.com/ethereum/go-ethereum/common"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	ethcrypto "github.com/ethereum/go-ethereum/crypto"
 )
 
-func Start(ctx context.Context) error {
-	jobs := []Job{
-		{
-			Name:    "Symbiotic",
-			Run:     newSymbioticRunner(),
-			Cadence: 1 * time.Hour,
-			Spend:   spend(tokens.WSTETH, ether1),
-		},
+func Start(
+	ctx context.Context,
+	xprovider xchain.Provider,
+	network netconf.Network,
+	rpcEndpoints xchain.RPCEndpoints,
+	keyPath string,
+) error {
+	if keyPath == "" {
+		return errors.New("private key is required")
+	}
+
+	privKey, err := ethcrypto.LoadECDSA(keyPath)
+	if err != nil {
+		return errors.Wrap(err, "load key", "path", keyPath)
+	}
+
+	backends, err := ethbackend.BackendsFromNetwork(network, rpcEndpoints, privKey)
+	if err != nil {
+		return err
+	}
+
+	var jobs []*types.Job
+
+	// Instantiate bridging of native ETH on a devnet
+	job, err := bridging.NewJob(
+		netconf.Devnet,
+		evmchain.IDMockL1,
+		evmchain.IDMockL2,
+		eoa.RoleMonitor,
+		common.Address{}, // native ETH
+		big.NewInt(0).Mul(util.MilliEther, big.NewInt(2)), // 0.002 ETH
+	)
+	if err != nil {
+		return err
+	}
+
+	jobs = append(jobs, job)
+
+	for _, job := range jobs {
+		go func() {
+			ticker := time.NewTicker(job.Cadence)
+			defer ticker.Stop()
+
+			for {
+				select {
+				case <-ctx.Done():
+					return
+				case <-ticker.C:
+					if err := run(ctx, xprovider, backends, job); err != nil {
+						log.Error(ctx, fmt.Sprintf("Job %s failed", job.Name), err)
+					}
+				}
+			}
+		}()
+	}
+
+	return nil
+}
+
+// run runs a job exactly once.
+func run(ctx context.Context, xprovider xchain.Provider, backends ethbackend.Backends, j *types.Job) error {
+	log.Debug(ctx, "Flowgen: running job", "name", j.Name)
+
+	// We fetch the latest finalized height of the source chain before we open the order.
+	// Once we open the order, we start fetching all event logs starting above this
+	// finalized height to retrieve the order status.
+	height, err := xprovider.ChainVersionHeight(ctx, xchain.ChainVersion{
+		ID:        j.SrcChain,
+		ConfLevel: xchain.ConfFinalized,
+	})
+	if err != nil {
+		return errors.Wrap(err, "chain height")
 	}
 
-	_ = jobs
-	_ = ctx
+	id, err := solvernet.OpenOrder(ctx, j.Network, evmchain.IDMockL1, backends, j.Owner, j.OrderData)
```

Comment:
We could consider making `OpenOrder` return the block height from the receipt. Then we wouldn't need to make the call above.

Reply:
you can add the order Id to ctx for simple logging: `ctx := log.WithCtx(ctx, "id", id)`